packages/react/src/components/dialog/Dialog.tsx

```diff
@@ -32,73 +32,34 @@ const findFocusableDialogElements = (dialogElement: HTMLElement): NodeList =>
     'a, button, textarea, input[type="text"], input[type="radio"], input[type="checkbox"], select',
   );
 
-const focusFirstDialogElement = (dialogElement?: HTMLElement) => {
-  if (dialogElement) {
-    const focusableElements = findFocusableDialogElements(dialogElement);
-
-    if (focusableElements.length) {
-      (focusableElements[0] as HTMLElement).focus();
-    }
-  }
-};
-
-const focusLastDialogElement = (dialogElement?: HTMLElement) => {
+const focusToDialogElement = (position: 'top' | 'bottom', dialogElement?: HTMLElement) => {
   if (dialogElement) {
     const focusableElements = findFocusableDialogElements(dialogElement);
     if (focusableElements.length) {
-      (focusableElements[focusableElements.length - 1] as HTMLElement).focus();
+      (focusableElements[position === 'top' ? 0 : focusableElements.length - 1] as HTMLElement).focus();
     }
   }
 };
 
-const addDocumentStartTabBarrier = (dialogElement?: HTMLElement): HTMLDivElement => {
-  const element = document.createElement('div');
-  element.className = 'hds-dialog-start-tab-barrier';
-  element.tabIndex = defaultBarrierProps.tabIndex;
-  element['aria-hidden'] = defaultBarrierProps.tabIndex['aria-hidden'];
-  element.addEventListener('focus', () => focusFirstDialogElement(dialogElement));
-  document.body.insertBefore(element, document.body.firstChild);
-  return element;
+const ContentTabBarrier = ({ onFocus }: { onFocus: () => void }): JSX.Element => {
+  /* eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex */
+  return <div {...defaultBarrierProps} onFocus={onFocus} />;
 };
 
-const addDocumentEndTabBarrier = (dialogElement?: HTMLElement): HTMLDivElement => {
+const addDocumentTabBarrier = (position: 'top' | 'bottom', dialogElement?: HTMLElement): HTMLDivElement => {
```

Comment:
A suggestion: this could be const or enum to avoid typing mistakes:
```ts
enum TabBarrierPosition {
  top = 'top',
  bottom = 'bottom'
}

if(TabBarrierPosition.top) {
 ...
}
```

Reply:
Yeah I'll do this quickly tomorrow 👍

Reply:
on iOS there's really somthing weird happing, sometimes it scrolls, sometimes it doesn't and sometimes it suddenly stops scrolling after it 1st started scrolling 🤦